lib/common/results.c

```diff
@@ -876,3 +876,47 @@ pcmk__reset_result(pcmk__action_result_t *result)
     free(result->action_stderr);
     result->action_stderr = NULL;
 }
+
+/*!
+ * \internal
+ * \brief Copy the result of an action
+ *
+ * \param[in]  src  Result to copy
+ * \param[out] dst  Where to copy \p src to
+ */
+void
+pcmk__copy_result(pcmk__action_result_t *src, pcmk__action_result_t *dst)
+{
```

Comment:
we've removed all cases where multiple results would reference the same string - right?
... and made that a design guideline.
so deep-copy is the only copy we'll ever need.

Reply:
Yes